### ISSUE TITLE
App crashed

### ISSUE THREAD

**Issue by aminalzanki** [google/iosched#112]
Here's the crash log. Please check :)

01-11 12:27:17.247  10949-10949/com.google.samples.apps.iosched E/AndroidRuntime﹕ FATAL EXCEPTION: main
    java.lang.RuntimeException: Unable to start activity ComponentInfo{com.google.samples.apps.iosched/com.google.samples.apps.iosched.ui.BrowseSessionsActivity}: android.content.res.Resources$NotFoundException: File res/drawable/selected_navdrawer_item_background.xml from drawable resource ID #0x7f0200eb
            at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2211)
            at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2261)
            at android.app.ActivityThread.access$600(ActivityThread.java:141)
            at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1256)
            at android.os.Handler.dispatchMessage(Handler.java:99)
            at android.os.Looper.loop(Looper.java:137)
            at android.app.ActivityThread.main(ActivityThread.java:5103)
            at java.lang.reflect.Method.invokeNative(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:525)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:737)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:553)
            at dalvik.system.NativeStart.main(Native Method)
     Caused by: android.content.res.Resources$NotFoundException: File res/drawable/selected_navdrawer_item_background.xml from drawable resource ID #0x7f0200eb
            at android.content.res.Resources.loadDrawable(Resources.java:2091)
            at android.content.res.Resources.getDrawable(Resources.java:695)
            at android.view.View.setBackgroundResource(View.java:14758)
            at com.google.samples.apps.iosched.ui.BaseActivity.formatNavDrawerItem(BaseActivity.java:1365)
            at com.google.samples.apps.iosched.ui.BaseActivity.makeNavDrawerItem(BaseActivity.java:1335)
            at com.google.samples.apps.iosched.ui.BaseActivity.createNavDrawerItems(BaseActivity.java:508)
            at com.google.samples.apps.iosched.ui.BaseActivity.populateNavDrawer(BaseActivity.java:486)
            at com.google.samples.apps.iosched.ui.BaseActivity.setupNavDrawer(BaseActivity.java:409)
            at com.google.samples.apps.iosched.ui.BaseActivity.onPostCreate(BaseActivity.java:541)
            at com.google.samples.apps.iosched.ui.BrowseSessionsActivity.onPostCreate(BrowseSessionsActivity.java:191)
            at android.app.Instrumentation.callActivityOnPostCreate(Instrumentation.java:1150)
            at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2194)
            at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2261)
            at android.app.ActivityThread.access$600(ActivityThread.java:141)
            at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1256)
            at android.os.Handler.dispatchMessage(Handler.java:99)
            at android.os.Looper.loop(Looper.java:137)
            at android.app.ActivityThread.main(ActivityThread.java:5103)
            at java.lang.reflect.Method.invokeNative(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:525)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:737)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:553)
            at dalvik.system.NativeStart.main(Native Method)
     Caused by: org.xmlpull.v1.XmlPullParserException: Binary XML file line #22: <item> tag requires a 'drawable' attribute or child tag defining a drawable
            at android.graphics.drawable.LayerDrawable.inflate(LayerDrawable.java:161)
            at android.graphics.drawable.Drawable.createFromXmlInner(Drawable.java:901)
            at android.graphics.drawable.Drawable.createFromXml(Drawable.java:837)
            at android.content.res.Resources.loadDrawable(Resources.java:2087)
            at android.content.res.Resources.getDrawable(Resources.java:695)
            at android.view.View.setBackgroundResource(View.java:14758)
            at com.google.samples.apps.iosched.ui.BaseActivity.formatNavDrawerItem(BaseActivity.java:1365)
            at com.google.samples.apps.iosched.ui.BaseActivity.makeNavDrawerItem(BaseActivity.java:1335)
            at com.google.samples.apps.iosched.ui.BaseActivity.createNavDrawerItems(BaseActivity.java:508)
            at com.google.samples.apps.iosched.ui.BaseActivity.populateNavDrawer(BaseActivity.java:486)
            at com.google.samples.apps.iosched.ui.BaseActivity.setupNavDrawer(BaseActivity.java:409)
            at com.google.samples.apps.iosched.ui.BaseActivity.onPostCreate(BaseActivity.java:541)
            at com.google.samples.apps.iosched.ui.BrowseSessionsActivity.onPostCreate(BrowseSessionsActivity.java:191)
            at android.app.Instrumentation.callActivityOnPostCreate(Instrumentation.java:1150)
            at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2194)
            at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2261)
            at android.app.ActivityThread.access$600(ActivityThread.java:141)
            at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1256)
            at android.os.Handler.dispatchMessage(Handler.java:99)
            at android.os.Looper.loop(Looper.java:137)
            at android.app.ActivityThread.main(ActivityThread.java:5103)
            at java.lang.reflect.Method.invokeNative(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:525)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:737)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:553)
            at dalvik.system.NativeStart.main(Native Method)

**Comment by eneim** [comment]
What have you done with your fork, in detail, the file res/drawable/selected_navdrawer_item_background.xml (do you have it or not)

and your BaseActivity. Please check again your call to formatNavDrawerItem() in BaseActivity.java. If you don't mind, please specify more about your crush: which kind of device you are using, which version of Android, Wifi or 3G/4G/LTE ... and you file BaseAcitivity.java as well as selected_navdrawer_item_background.xml in drawable folder.

**Comment by eneim** [comment]
FYI, supposing use have this in your repo:
https://github.com/google/iosched/blob/master/android/src/main/res/drawable/selected_navdrawer_item_background.xml

And the line where you app crush is 22nd line, where <item android:drawable ... /> is defined.

If you are using a low-version (Version Code < 11), the attribute ?android:selectableItemBackground may not available (ref here: http://developer.android.com/reference/android/R.attr.html#selectableItemBackground). Please consider to modify your code (use your custom drawable for example)

**Comment by eeVoskos** [comment]
The app seems to crash in pre-Lollipop devices (tested on Android 4.4, 4.3). Changing lines 1364-1366 in `BaseActivity.java`, as the snippet below, prevents the crash:
```
if (selected && Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
    view.setBackgroundResource(R.drawable.selected_navdrawer_item_background);
}
```

However, it doesn't produce the same result in the UI (the selected navigation item doesn't have the correct background). The used drawable needs to be changed appropriately.

**Comment by romannurik** [comment]
Looks like this is my fault. I'm on vacation so won't be able to push a fix for a few weeks. PRs welcome!

**Comment by Merguez22** [comment]
I have the same error on a Samsung S4 with Android 4.4.2

**Comment by andrewaarestad** [comment]
eeVoskos's snippet fixed the crash for me on Samsung S2 Android 4.4.3.

**Comment by tjohns** [comment]
This should be fixed by commit dd7ed72a7eb2d223203db079bd99d31c6ef3061e. Thanks!
